client/state/push-notifications/actions.js

```diff
@@ -37,12 +37,20 @@ import {
 	isServiceWorkerSupported,
 	registerServerWorker,
 } from 'lib/service-worker';
+import {
+	recordTracksEvent
+} from 'state/analytics/actions';
 
 const debug = debugFactory( 'calypso:push-notifications' );
 const DAYS_BEFORE_FORCING_REGISTRATION_REFRESH = 15;
 const serviceWorkerOptions = {
 	path: '/service-worker.js',
 };
+const tracksEvent = ( dispatch, eventName, props ) => {
```

Comment:
If you look at what ultimately [gets executed](https://github.com/Automattic/wp-calypso/blob/673fd7cced1d5c8ff75ce47ecf3cf24b8c279455/client/lib/analytics/index.js#L122-L147), it doesn't block for very long, so the timeout is a pretty small optimization here, IMO.

I'd suggest if we are going to use the timeout, I'd add an action called `asyncRecordTracksEvent` to the [analytics actions](https://github.com/Automattic/wp-calypso/blob/673fd7cced1d5c8ff75ce47ecf3cf24b8c279455/client/state/analytics/actions.js#L58) module to keep from passing `dispatch` around.